File: 0861-flipping-an-image/0861-flipping-an-image.cpp
class Solution {
public:
    vector<vector<int>> flipAndInvertImage(vector<vector<int>>& image) {
        for (vector<int>& images: image) {
            std::ranges::reverse(images);
            for (int& i: images) {
                i = i == 0 ? 1 : 0;
            }
        }

        return image;
    }
};

File: 0861-flipping-an-image/0861-flipping-an-image.java
class Solution {
    public int[][] flipAndInvertImage(int[][] image) {
        for (int[] images: image) {
            int start = 0, end = images.length-1;
            while (start < end) {
                images[start] = (images[start] + images[end]) - (images[end] = images[start]);
                start++;
                end--;
            }

            for (int i = 0; i < images.length; i++) {
                images[i] = images[i] == 0 ? 1 : 0;
            }
        }

        return image;
    }
}